pkg/cloudprovider/providers/azure/azure_zones.go

```diff
@@ -61,14 +62,36 @@ func (az *Cloud) GetZone() (cloudprovider.Zone, error) {
 // This is particularly useful in external cloud providers where the kubelet
 // does not initialize node data.
 func (az *Cloud) GetZoneByProviderID(providerID string) (cloudprovider.Zone, error) {
-	return cloudprovider.Zone{}, errors.New("GetZoneByProviderID not implemented")
+	nodeName, err := splitProviderID(providerID)
+	if err != nil {
+		return cloudprovider.Zone{}, err
+	}
+	return az.GetZoneByNodeName(nodeName)
 }
 
 // GetZoneByNodeName implements Zones.GetZoneByNodeName
 // This is particularly useful in external cloud providers where the kubelet
 // does not initialize node data.
 func (az *Cloud) GetZoneByNodeName(nodeName types.NodeName) (cloudprovider.Zone, error) {
-	return cloudprovider.Zone{}, errors.New("GetZoneByNodeName not imeplemented")
+
+	vm, err := az.VirtualMachinesClient.Get(az.ResourceGroup, string(nodeName), compute.InstanceView)
+
+	exists, err := checkResourceExistsFromError(err)
+	if err != nil {
+		return cloudprovider.Zone{}, err
+	}
+
+	if !exists {
+		return cloudprovider.Zone{}, nil
+	}
```

Comment:
Node non-existence should be an error in this case. There's no need to call checkResourceExistsFromError for a check.